### PR TITLE
Update sigh.js to detect when peg & railroad steps can be skipped.

### PR DESCRIPTION
**Pull request by mykmartin**
This allows the watch command to monitor files in runtime with infinite
looping.

### CODE REVIEW

File: tools/sigh.js [hunk 1]
```diff
@@ -89,45 +113,37 @@ function railroad() {
...
-    };
-
-    let output = renderTemplate(data, path.resolve(projectRoot, path.join('node_modules', 'grammkit', 'template', 'viewer.html')));
+    return {name, rules};
+  });
 
-    fs.writeFileSync(path.resolve(projectRoot, railroadFile), output);
-  }
+  let data = {
+      title: `Railroad diagram for ${sources.peg.grammar}`,
```
Comment (dstockwell) [inline tools/sigh.js]: normally just a 2 space indent here

**Comment by dstockwell** [comment]
Surely these things are sane on windows too :man_shrugging:

File: tools/sigh.js (hunk 1 above)
Comment (mykmartin) [reply tools/sigh.js]: Fixed (once I work out how to get the new commit in)

**Comment by mykmartin** [comment]
> Surely these things are sane on windows too

Should be fine; the unix path notation is used elsewhere and fixed by the path module, and node's docs on fs doesn't mention any OS-specific things for stat-ing files.

**Pull request merged**